Tests/SwerverTests/RouterSpec/RouterSpec.swift

```diff
@@ -6,121 +6,212 @@ import Foundation
 class RouterSpec: QuickSpec {
     override func spec() {
         describe("A Router") {
-            var router = Router()
-
-            guard let urlOne = URL(string: Resource.test_get.rawValue),
-                let urlTwo = URL(string: Resource.test_head.rawValue),
-                let urlThree = URL(string: Resource.test_option.rawValue),
-                let urlNotFound = URL(string: "/not_here") else {
-                return
+            func buildRouter() -> Router {
+                return Router()
             }
-
-            context("when receive an invalid request object") {
-                let anInvalidRequest: HttpRequest? = nil
-                let expectedResponse = router.process(request: anInvalidRequest)
-                
+            
+            context("receives an invalid request object") {
                 it("returns a response with a 404 status code") {
-                    expect(expectedResponse.statusCode).to(equal(StatusCode.not_found.rawValue))
+                    let router = buildRouter()
+                    let anInvalidRequest: HttpRequest? = nil
+                    
+                    let response = router.process(request: anInvalidRequest)
+                    let expectedStatusCode = response.statusCode
+                    
+                    expect(expectedStatusCode).to(equal(StatusCode.not_found.rawValue))
                 }
                 
                 it("returns a response with a content type of text") {
-                    expect(expectedResponse.get(header: .contentType)).to(equal(ContentType.text.rawValue))
-                }
-                
-                it("returns a response with an empty body") {
-                    expect(expectedResponse.get(header: .contentLength)).to(equal("0"))
+                    let router = buildRouter()
+                    let anInvalidRequest: HttpRequest? = nil
+                    
+                    let response = router.process(request: anInvalidRequest)
+                    let expectedContentType = response.get(header: .contentType)
+                    
+                    expect(expectedContentType).to(equal(ContentType.text.rawValue))
                 }
             }
-
-            context("when receive a GET request for an existing route") {
-                let aGetRequest = HttpRequest(method: RequestMethod.get, url: urlOne)
-                let expectedResponse = router.process(request: aGetRequest)
-                
+            
+            context("receives a GET request for an existing route") {
                 it("returns an response with a 200 status code") {
-                    expect(expectedResponse.statusCode).to(equal(StatusCode.ok.rawValue))
+                    let router = buildRouter()
+                    let targetRoute = TestData.simpleGetUrl()
```

Comment:
This could probable just be `TestData.validUrlOne()`. There is not a compelling reason to add additional functions to `TestData`.

Reply:
will fix that

Reply:
fixed